pkg/test/masteruserrecord/doubles.go

```diff
@@ -0,0 +1,145 @@
+package murtest
+
+import (
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/redhat-cop/operator-utils/pkg/util"
+	uuid "github.com/satori/go.uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+type MurModifier func(mur *toolchainv1alpha1.MasterUserRecord)
+type UaInMurModifier func(targetCluster string, mur *toolchainv1alpha1.MasterUserRecord)
+
+func NewMasterUserRecord(userName string, modifiers ...MurModifier) *toolchainv1alpha1.MasterUserRecord {
+	userId := uuid.NewV4().String()
+	mur := &toolchainv1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      userName,
+			Namespace: test.HostOperatorNs,
+		},
+		Spec: toolchainv1alpha1.MasterUserRecordSpec{
+			UserID:       userId,
+			UserAccounts: []toolchainv1alpha1.UserAccountEmbedded{newEmbeddedUa(test.MemberClusterName, userId)},
+		},
+	}
+	ModifyMur(mur, modifiers...)
+	return mur
+}
+
+func newEmbeddedUa(targetCluster, userId string) toolchainv1alpha1.UserAccountEmbedded {
+	return toolchainv1alpha1.UserAccountEmbedded{
+		TargetCluster: targetCluster,
+		SyncIndex:     "123abc",
+		Spec: toolchainv1alpha1.UserAccountSpec{
+			UserID:  userId,
+			NSLimit: "basic",
+			NSTemplateSet: toolchainv1alpha1.NSTemplateSetSpec{
+				TierName: "basic",
+				Namespaces: []toolchainv1alpha1.Namespace{
+					{
+						Type:     "ide",
+						Revision: "123abc",
+						Template: "",
+					},
+					{
+						Type:     "ci/cd",
+						Revision: "123abc",
+						Template: "",
+					},
+					{
+						Type:     "staging",
+						Revision: "123abc",
+						Template: "",
+					},
+				},
+			},
+		},
+	}
+}
+
+func ModifyMur(mur *toolchainv1alpha1.MasterUserRecord, modifiers ...MurModifier) {
```

Comment:
should we make this function private?

Reply:
This one is not used, indeed. But I guess that for consistency it would be good to keep it exported as `ModifyUa` is - so it can be used in a test.

Reply:
ok, makes sense. Since it's in the `masteruserrecord` pkg, could you rename this func to `Modify`? (no need to repeat the `Mur` here)

Reply:
done 63785f9